From 9741e102e1b3b4cd45babe8f194a34b469fdc632 Mon Sep 17 00:00:00 2001
From: Jan Kiszka <jan.kiszka@siemens.com>
Date: Thu, 27 Sep 2018 08:35:37 +0200
Subject: [PATCH 11/21] drivers/analogy: Fix improper memset in Intel 8255

This overwrote the pointer itself - and more. Long pending bugs.

Signed-off-by: Jan Kiszka <jan.kiszka@siemens.com>
---
 kernel/drivers/analogy/intel/8255.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/kernel/drivers/analogy/intel/8255.c b/kernel/drivers/analogy/intel/8255.c
index ad8039b..1abe250 100644
--- a/kernel/drivers/analogy/intel/8255.c
+++ b/kernel/drivers/analogy/intel/8255.c
@@ -201,7 +201,7 @@ void a4l_subdev_8255_init(struct a4l_subdevice *subd)
 {
 	subd_8255_t *subd_8255 = (subd_8255_t *)subd->priv;
 	/* Initializes the subdevice structure */
-	memset(&subd, 0, sizeof(struct a4l_subdevice));
+	memset(subd, 0, sizeof(struct a4l_subdevice));
 
 	/* Subdevice filling part */
 
@@ -263,7 +263,7 @@ static int dev_8255_attach(struct a4l_device *dev, a4l_lnkdesc_t *arg)
 			goto out_attach;
 		}
 
-		memset(&subd, 0, sizeof(struct a4l_subdevice));
+		memset(subd, 0, sizeof(struct a4l_subdevice));
 		memset(subd->priv, 0, sizeof(subd_8255_t));
 
 		subd_8255 = (subd_8255_t *)subd->priv;
-- 
1.9.1

